test/unit/DCAStrategies/dca-strategies-management-handler.spec.ts

```diff
@@ -0,0 +1,106 @@
+
+  before('Setup accounts and contracts', async () => {
+    const factory: DCAStrategiesManagementHandlerMock__factory = await ethers.getContractFactory('DCAStrategiesManagementHandlerMock');
+    DCAStrategiesManagementHandlerMock = await factory.deploy();
+    snapshotId = await snapshot.take();
+    chainId = BigNumber.from((await ethers.provider.getNetwork()).chainId);
+  });
+
+  beforeEach(async () => {
+    await snapshot.revert(snapshotId);
+  });
+
+  describe('constructor', () => {
+    when('handler is deployed', () => {
+      then('strategy counter is correct', async () => {
+        const strategyCounter = await DCAStrategiesManagementHandlerMock.strategyCounter();
+        expect(strategyCounter).to.equal(0);
+      });
+    });
+  });
+
+  describe('createStrategy', () => {
+    const OWNER = wallet.generateRandomAddress();
+    const NAME = ethers.utils.formatBytes32String('Optimism Ecosystem - v1');
+    const SHARE_TOKEN_A = { token: wallet.generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARE_TOKEN_B = { token: wallet.generateRandomAddress(), share: BigNumber.from(50) };
+    const SHARES = [SHARE_TOKEN_A, SHARE_TOKEN_B];
+
+    when('owner is zero address', () => {
+      then('tx reverted with message', async () => {
+        await expect(DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, constants.ZERO_ADDRESS)).to.be.revertedWith(
+          'ZeroAddress()'
+        );
+      });
+    });
+    when('strategy is created', () => {
+      let tx: TransactionResponse;
+      let strategy: IDCAStrategiesManagementHandler.StrategyStruct;
+      let tokenShares: IDCAStrategies.ShareOfTokenStruct[];
+      given(async () => {
+        tx = await DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, OWNER);
+        strategy = await DCAStrategiesManagementHandlerMock.getStrategy(1);
+        tokenShares = strategy.tokens;
+      });
+      when('name already exists', () => {
+        then('tx reverted with message', async () => {
+          await expect(DCAStrategiesManagementHandlerMock.createStrategy(NAME, SHARES, OWNER)).to.be.revertedWith('NameAlreadyExists()');
+        });
+      });
+      then('counter is correct', async () => {
+        expect(await DCAStrategiesManagementHandlerMock.strategyCounter()).to.be.equal(1);
+      });
+      then('owner is correct', async () => {
+        expect(strategy.owner).to.be.equal(OWNER);
+      });
+      then('name is correct', async () => {
+        expect(strategy.name).to.be.equal(NAME);
+      });
+      then('name is saved correctly in reversed mapping', async () => {
+        expect(await DCAStrategiesManagementHandlerMock.strategyIdByName(NAME)).to.be.equal(1);
+      });
+      then('version is correct', async () => {
+        expect(strategy.currentVersion).to.be.equal(1);
+      });
+      then('shares are correct', async () => {
+        SHARES.forEach((s, i) => {
```

Comment:
You should also check that `SHARES.length === strategy.tokenShares.length`. Otherwise, the test could pass but `strategy.tokenShares` could have more tokens